docs: add driver description and demo (#3142)
graycoreio/daffodil: libs/product/guides/drivers.md

    }
  }
}
```

## Dynamic driver switching

> Drivers act as bridges between an app and its backend, making it easier to integrate different services in an application. Instead of requiring developers to understand the specific details of each system, they can instead interact with them through a driver. This promotes flexibility in data management while providing a consistent way to communicate with each service.

Reviewer: ```suggestion
Drivers act as bridges between an application and its backend, simplifying the integration of different platforms. Instead of needing to understand the specific details of each platform, you can interact with them through a driver. This promotes flexibility in data management while providing a consistent way to communicate with each platform.
```

> The demo [here](https://stackblitz.com/edit/stackblitz-starters-nwvydt?file=src%2Fmain.ts) showcases an application with dynamic driver switching, which allows product information from different drivers to become accessible from one URL.

Reviewer: ```suggestion
This [demo](https://stackblitz.com/edit/stackblitz-starters-nwvydt?file=src%2Fmain.ts) showcases an application with dynamic driver switching, which allows product information from different drivers to become accessible from one URL.
```

    }
  }
}
```

## Dynamic driver switching

> 

Reviewer 2: I don't think we want to direct people to use the dynamic switch driver themselves. Instead I think we should use it to highlight how the drivers work.
